e2e/tests/node_not_ready/node_power_off.robot

```diff
@@ -0,0 +1,491 @@
+*** Settings ***
+Documentation       Test the Longhorn resillence if cluster node powering off
+
+Resource            ../../keywords/common.resource
+Resource            ../../keywords/engine.resource
+Resource            ../../keywords/replica.resource
+Resource            ../../keywords/node.resource
+Resource            ../../keywords/volume.resource
+
+Suite Setup         set_test_suite_environment
+Test Setup          set_test_environment    ${TEST NAME}
+Test Teardown       Cleanup resource and resume state
+
+
+*** Variables ***
+${Gi}=                  2**30
+${volume_size_gb}=      1
+${sleep_interval}=      300
+${volume_type}=         RWO
+
+
+*** Test Cases ***
+Node power off with replica-VA
+    ${number_of_replicas}=    Convert To Integer    3
+    Set Test Variable    ${number_of_replicas}
+
+    ${volume_type}=    Evaluate    "${volume_type}".lower()
+
+    ${field1}=    Convert To String    {"spec": {"size": "${${volume_size_gb} * ${Gi}}"}}
+    ${field2}=    Convert To String    {"spec": {"numberOfReplicas": ${number_of_replicas}}}
+    ${field3}=    Convert To String    {"spec": {"accessMode": "${volume_type}"}}
+    @{list_of_fields}=    Create List    ${field1}    ${field2}    ${field3}
+
+    Given Create Volume With Fields    ${list_of_fields}
+    And Attach volume to node 1
+    And Write data into mount point
+
+    When Power off node 1
+
+    Then Node 1 should have 1 volume replica
+    And Node 1 state should eventually be NotReady
+    And Volume state should eventually be unknown
+    And Engine state should eventually be unknown
+    And Replica state on node 1 should eventually be unknown
```

Comment:
Should we check for all replicas?

Reply:
I think you are right. So I check all test cases and add check replica status step under each case.https://github.com/longhorn/longhorn-tests/blob/ccca467468a21e0181a592e0d238aefa1a9a3424/e2e/tests/node_not_ready/node_power_off.robot#L44-L46